### PR TITLE
Add upgrade test

### PR DESCRIPTION
**Pull request by berendt**
OpenStack Xena -> OpenStack Yoga
Ceph Pacific -> Ceph Pacific

Signed-off-by: Christian Berendt <berendt@osism.tech>

### CODE REVIEW

File: playbooks/upgrade.yml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    - name: Run part 1 + 2
+      shell:
+        chdir: "{{ basepath }}/ansible"
+        cmd: |
+          {{ ansible_playbook }} -i ../terraform/inventory.{{ cloud_env }} --key-file .id_rsa.{{ cloud_env }} manager-part-1.yml -e cloud_env={{ cloud_env }} -e repo_path={{ repo_path }}
+
+    - name: Run part 3
+      shell:
+        chdir: "{{ basepath }}/ansible"
+        cmd: "ssh -i .id_rsa.{{ cloud_env }} dragon@{{ manager_host }} /opt/configuration/ansible/manager.sh"
```
Comment (osfrickler) [inline playbooks/upgrade.yml:73]: s-ansible/manager.sh-scripts/deploy.sh-

File: .zuul.yaml [hunk 2]
```diff
@@ -25,6 +44,7 @@
         - tox-docs
         - tox-linters
         - testbed-deploy
```
Comment (osfrickler) [inline .zuul.yaml:46]: Maybe comment out the other jobs while testing for faster turnaround

File: playbooks/upgrade.yml [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+      shell:
+        chdir: "{{ basepath }}/ansible"
+        cmd: "ssh -i .id_rsa.{{ cloud_env }} dragon@{{ manager_host }} /opt/configuration/scripts/deploy.sh"
+      when: not manual_deploy
+
+    - name: Run checks before the upgrade
+      shell:
+        chdir: "{{ basepath }}/terraform"
+        cmd: make ENVIRONMENT={{ cloud_env }} check
+      when: not manual_deploy
```
Comment (osfrickler) [inline playbooks/upgrade.yml:80]: Can you just import the deploy playbook for everything up to this? Would avoid a lot of duplication.

File: doc/source/usage.rst [hunk 4]
```diff
@@ -38,47 +38,47 @@ Deploy services
 On the testbed, the services can currently be deployed manually. In the future, these manual
 steps will be automated by Zuul CI.
 
-* Basic infrastructure services (MariaDB, RabbitMQ, Redis, ...)
+* Ceph services
 
   .. code-block:: console
 
-     /opt/configuration/scripts/002-infrastructure-services-basic.sh
+     /opt/configuration/scripts/deploy/002-ceph-services.sh
```
Comment (osfrickler) [inline doc/source/usage.rst:45]: Maybe we can renumber the scripts when we rename them anyway to keep them ordered and have enough space for additions. This one would then become e.g. 100-ceph-services.sh ...

File: doc/source/usage.rst [hunk 5]
```diff
@@ -38,47 +38,47 @@ Deploy services
...
-     /opt/configuration/scripts/006-infrastructure-services-extended.sh
+     /opt/configuration/scripts/deploy/003-infrastructure-services-basic.sh
 
-* Ceph services
+* Extended infrastructure services (Patchman, phpMyAdmin, ...)
 
   .. code-block:: console
 
-     /opt/configuration/scripts/003-ceph-services.sh
+     /opt/configuration/scripts/deploy/006-infrastructure-services-extended.sh
```
Comment (osfrickler) [inline doc/source/usage.rst:57]: These two 200-infra* and 210-infra*

File: doc/source/usage.rst [hunk 6]
```diff
@@ -38,47 +38,47 @@ Deploy services
...
 
 * Basic OpenStack services (Compute, Storage, ...)
 
   .. code-block:: console
 
-     /opt/configuration/scripts/004-openstack-services-basic.sh
+     /opt/configuration/scripts/deploy/004-openstack-services-basic.sh
```
Comment (osfrickler) [inline doc/source/usage.rst:63]: These 3xx-openstack-* etc.

**Comment by berendt** [comment]
@osfrickler Works now. I have reactivated the other jobs.

**Review by osfrickler — APPROVED** [review]

**Pull request merged**